[#245] 글쓰기: 수정 기능 구현

jwonyLee opened the pull request:
## 개요

![ezgif-2-76925612f22f](https://user-images.githubusercontent.com/15073405/143732899-8eb4583b-4917-4189-8a4c-9fdb1b599f18.gif)

글쓰기: 수정 기능 구현

## 작업 사항

- 게시물 수정 기능 구현
- 게시물 수정일 때 실제 데이터를 글쓰기 입력폼에 표시
- 게시물에서 수정 버튼을 선택하면 게시물 수정화면으로 이동, 저장 시 샀어요로 저장
- 게시물에서 카테고리가 제대로 표시 되고 있지 않아서.. 이미지와 동일하게 reloadData() 호출
- 사고싶다 글쓰기일 때 만족도가 표시되면 안되는데, 표시되고 있어서 수정했습니다.

### Linked Issue

close #245

jwonyLee committed:
[#245] feat: 게시물에서 수정 버튼을 선택하면 게시물 수정화면으로 이동
- 이 때 수정할 PostEntity를 같이 전달한다.

jwonyLee committed:
[#245] feat: 게시물 수정일 때 실제 데이터를 입력폼에 표시

jwonyLee committed:
[#245] feat: 게시물 수정 기능 구현

jwonyLee committed:
[#245] feat: 샀어요 버튼 선택 시 글쓰기 화면으로 이동
- 카테고리가 제대로 표시되지 않는 문제 해결

gustn3965 commented on ThingLog/ViewModel/WriteViewModel.swift:
```diff
@@ -130,6 +145,37 @@ final class WriteViewModel {
+        modifyEntity.price = Int64(typeValues[1]?.filter("0123456789".contains) ?? "") ?? 0
+        modifyEntity.purchasePlace = typeValues[2]
+        modifyEntity.postType?.pageType = .bought
+
```
사고싶다에서 게시글을 작성한뒤에, 수정기능을 누른다음에, 선물준사람에 내용이 비어있는 경우에 `typeValues[2]` 이부분에서 인덱스 에러가 나고 있습니당. 

------------ 

그리고 `typeValues` 배열 인덱싱으로, `title`, `price`, `purchagePlace`를 지정하는게 약간 가독성에는 좋지 못한 것 같습니다. 이부분은 나중에 리팩토링하셔도 좋을 것 같습니다!

gustn3965 commented on ThingLog/View/TableVeiwCell/PostTableCell/PostTableCell.swift:
```diff
@@ -349,6 +349,7 @@ extension PostTableCell {
         if let categories: [CategoryEntity] = categories?.allObjects as? [CategoryEntity] {
             categoryViewDataSource.items = categories.isEmpty ? ["카테고리"] : categories.compactMap { $0.title }
             categoryViewDataSource.isEmpty = categories.isEmpty
+            categoryCollectionView.reloadData()
         }
     }
 
```
이 코드는 아니지만, `PostTableCell` 의 `configurePriceLabel` 메소드에서, `priceLabel`의 색을 `.black`으로 지정하고 있어서, 다크모드에서는 보이지 않고 있습니다 🥲

gustn3965 commented on ThingLog/ViewController/Post/PostViewController+DataSource.swift:
```diff
@@ -18,9 +18,31 @@ extension PostViewController: UITableViewDataSource {
+        }
+
+        return cell
+    }
```
코드에는 문제가 없는 것 같은데, 다만 셀내의 뷰들을 `bind**`로 메서드를 통일성있게 한 반면,` moreMenuButton`은 따로 있어서 아쉽습니당. 🥲  물론 rxSwift를 사용하지 않긴 하지만, 나름 이벤트를 작성하고 있어서, bind 메서드명이 아니더래도 통일성있게 따로 메소드로 뺴셔도 좋을 것 같다는 생각입니당.  ( 반영 안하셔도 됩니다! )

gustn3965 commented on ThingLog/ViewController/Write/WriteViewController+DataSource.swift:
```diff
@@ -104,6 +106,10 @@ extension WriteViewController {
         }
 
         cell.delegate = self
+        if viewModel.contents.isNotEmpty {
+            cell.textView.text = viewModel.contents
+            cell.textView.textColor = SwiftGenColors.primaryBlack.color
+        }
```
이부분은 한쪽에만 설정하도록 되어있어서, 재사용할 때 이슈가 있을것으로 생각됩니다. `empty`인경우에는 빈 값을 할당하거나, 아니면 `prepareForReuse`에서 빈값을 할당해야할 것 같습니다.

gustn3965 reviewed (approved):
우선 리뷰가 늦어진 점 죄송합니다.🙏🏻
------------- 

LGTM 고생하셨습니다!  

코멘트 확인 및 반영하시고 머지하시면 될 것 같습니다 !

jwonyLee committed:
[#245] ref: 피드백 반영
- 게시물에서 가격 색상 변경
- moreMenuButton callback 설정 코드 별도 메소드로 분리
- 글쓰기 화면에서 자유 글쓰기 입력 기본 값 구성 변경
- updatePost() 수정

jwonyLee commented:
코멘트 남겨주신 부분 전부 반영했습니다! `typeValues` 부분은 조금 더 고민해보고 별도로 리팩토링 진행하겠습니다~
